### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by AlejandraKeber**
## Hello code reviewer! 🌤️
In this milestone of the project I did the following:

## Changes made: ⛏️

- [ ] Remove current_user method in ApplicationController, devise will provide us with one.
- [ ] Install and setup devise:
1. Can register a new user.
2. User logs in with a combination of email and password.
3. Hashed password should be stored in the database.
4. Ask for confirmation of email.
5. Can reset password.
- [ ] Modify the views in devise for registration and login to match the wireframes from the Sneak Peeke and my styling.

Thanks!!! 👋 👋 👋

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
```
Comment (Lucas-Masaba) [inline app/models/user.rb:4]: - [ ] As mentioned in the project requirements, when users sign up, they should be prompted to confirm their emails. Please add `:confirmable` to this list.

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @AlejandraKeber  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Well-written pull request message :writing_hand: 
- Successfully added devise :gear: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (AlejandraKeber) [reply app/models/user.rb:4]: Thanks for your advice! I will add it in the next commit

File: app/models/user.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@ class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
+         :recoverable, :rememberable, :validatable, :confirmable
```
Comment (zieeco) [inline app/models/user.rb:5]: - Hey, great job fixing the issue suggested by the previous reviewer, however, you need to update the migration file for the `add_devise_to_user` by uncommenting these [lines of code](https://github.com/AlejandraKeber/BlogApp/blob/461b77dab7e31893b2c7fa22048a9edbc4b1c5ec/db/migrate/20230620205519_add_devise_to_users.rb#L23-L26), and [these also](https://github.com/AlejandraKeber/BlogApp/blob/461b77dab7e31893b2c7fa22048a9edbc4b1c5ec/db/migrate/20230620205519_add_devise_to_users.rb#L37-L39) to make it fully functional. Please, kindly fix this also.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @AlejandraKeber,





Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the changes, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice adding the devise. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions
- N/A


**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 2 above)
Comment (AlejandraKeber) [reply app/models/user.rb:5]: Hi @zieeco I don't know what you are asking for.  I already update the migration file for add_devise_to user to add confirmable:
image
I also add :confirmable and the app is working fine and asking for confirmation when you sign up .... are you asking me to modify a commented line in the User model (line 3)?

Comment (zieeco) [reply app/models/user.rb:5]: @AlejandraKeber, You don't need to create a separate migration file for this. I highlighted the piece of code you should uncomment in `db/migrate/20230620205519_add_devise_to_users.rb`. Please, check these [lines of code](https://github.com/AlejandraKeber/BlogApp/blob/461b77dab7e31893b2c7fa22048a9edbc4b1c5ec/db/migrate/20230620205519_add_devise_to_users.rb#L23-L26), and [these also](https://github.com/AlejandraKeber/BlogApp/blob/461b77dab7e31893b2c7fa22048a9edbc4b1c5ec/db/migrate/20230620205519_add_devise_to_users.rb#L37-L39) and uncomment them.

**Review by fobadara — APPROVED** [review]
Hi @AlejandraKeber, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Proper implementation of the reviewer's suggestions

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**